data_handler/handlers/liquidable_debt/debt_handlers.py

```diff
@@ -0,0 +1,190 @@
+                                             HEALTH_FACTOR_FIELD_NAME, DEBT_USD_FIELD_NAME, FIELDS_TO_VALIDATE,
+                                             ALL_NEEDED_FIELDS, LIQUIDABLE_DEBT_FIELD_NAME)
+from handlers.loan_states.zklend.events import ZkLendState
+from handlers.helpers import TokenValues
+
+
+class GCloudLiquidableDebtDataHandler:
+    """
+    A handler that collects data from Google Cloud Storage bucket,
+    parses it and stores it in the database.
+
+    :cvar AVAILABLE_PROTOCOLS: A list of all available protocols.
+    :method: `update_data` -> Updates the data stored in the database.:
+    """
+    AVAILABLE_PROTOCOLS = [item.value for item in LendingProtocolNames]
+    CONNECTOR = DBConnector()
+
+    def __init__(
+            self,
+            loan_state_class: State,
+            connection_url: str = GS_BUCKET_URL,
+            bucket_name: str = GS_BUCKET_NAME,
+            collector: Collector = GoogleCloudDataCollector
+    ):
+        self.collector = collector
+        self.connection_url = connection_url
+        self.bucket_name = bucket_name
+        self.state_class = loan_state_class
+
+    def prepare_data(self, protocol_name: str, path: str = LOCAL_STORAGE_PATH) -> dict:
+        uploaded_file_path = self.collector.collect_data(
+            protocol_name=protocol_name,
+            available_protocols=self.AVAILABLE_PROTOCOLS,
+            bucket_name=self.bucket_name,
+            path=path,
+            url=self.connection_url
+        )
+        parsed_data = self._parse_file(uploaded_file_path)
+
+        return self._calculate_liquidable_debt(parsed_data)
+
+    def _calculate_liquidable_debt(self, data: dict = None):
+        """
+        Calculates liquidable debt based on data provided and updates an existing data.
+        :param data: Data to calculate liquidable debt for.
+        :return: A dictionary of the ready liquidable debt data.
+        """
+        result_data = deepcopy(data)
+
+        for row_number in data:
+            state = self.state_class(verbose_user=data[row_number][USER_FIELD_NAME])
+
+            state.loan_entities[data[row_number][USER_FIELD_NAME]].debt.values = {
+                key: value
+                for key, value in data[row_number][DEBT_FIELD_NAME].items()
+            }
+            state.loan_entities[data[row_number][USER_FIELD_NAME]].collateral.values = {
+                key: value
+                for key, value in data[row_number][COLLATERAL_FIELD_NAME].items()
+            }
+
+            for token in data[row_number][DEBT_FIELD_NAME]:
+                if not data[row_number][COLLATERAL_FIELD_NAME].get(token, ""):
+                    continue
+
+                result = state.compute_liquidable_debt_at_price(
+                    prices=TokenValues(),
+                    collateral_token=token,
+                    collateral_token_price=data[row_number][COLLATERAL_FIELD_NAME][token],
```

Comment:
This is a hypothetical price which we take from a range of values between 0 and 1.3 * current price. See e.g. here: https://github.com/CarmineOptions/derisk-research/blob/a0d7c0172825174c40aa388212d103daa0e53097/data_handler/handlers/order_books/uniswap_v2/main.py#L43. cc @djeck1432 

The goal is to iterate over each price level and compute the liquidable debt for each price level.